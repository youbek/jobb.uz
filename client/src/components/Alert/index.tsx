import React from "react";
import styled from "styled-components";

const Container = styled.div`
  padding: 4rem;
  font-weight: 500;
  text-align: center;
  margin-top: 2rem;
  background-color: #f0f0f0;
  color: #707070;
  border-radius: 10px;
`;

interface Props extends React.ComponentProps<"div"> {}

function Alert(props: Props) {
  const { children } = props;

  return <Container>{children}</Container>;
}

export default Alert;
