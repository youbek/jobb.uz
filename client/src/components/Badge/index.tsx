import React from "react";
import styled, { css } from "styled-components";

interface Props {
  color: "green" | "blue";
}

const green = css`
  color: #468a74;
  border: 1px solid #468a74;
`;

const blue = css`
  color: #148c96;
  border: 1px solid #148c96;
`;

const Badge = styled.span<Props>`
  display: inline-flex;
  align-items: center;
  font-size: 12px;
  font-weight: 700;
  color: #cc3d86;
  box-sizing: border-box;
  padding: 4px 10px;
  margin-right: 5px;
  border: 1px solid #cc3d86;
  border-radius: 4px;

  ${({ color }) => color === "green" && green}
  ${({ color }) => color === "blue" && blue}
`;

export default Badge;
