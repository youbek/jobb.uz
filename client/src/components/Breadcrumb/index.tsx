import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";

const StyledBreadcrumb = styled.div`
  display: flex;
  flex-wrap: wrap;
  background-color: #2c3038;
  border-bottom: 1px solid #eee;
  padding-right: 15px;
  padding-left: 15px;
  padding-top: 10px;
  margin-bottom: 2rem;
  margin-top: 0;
  color: #fff9;
  font-weight: 600;
  border-top: 1px solid #3c3c3c;
  list-style: none;
`;

const StyledBreadcrumb__Item = styled.div`
  display: inline-block;
  position: relative;
  padding-bottom: 10px;
  font-size: 14px;

  a {
    color: #fff9;

    &:hover {
      color: #fff9;
      text-decoration: none;
    }
  }

  & + & {
    padding-left: 0.5rem;
  }

  & + &:before {
    display: inline-block;
    padding-right: 0.5rem;
    color: #6c757d;
    content: "/";
  }
`;

interface Props {
  categoryName?: string;
}

function Breadcrumb({ categoryName }: Props) {
  return (
    <StyledBreadcrumb>
      <StyledBreadcrumb__Item>
        <Link to="/">Работа в Ташкенте</Link>
      </StyledBreadcrumb__Item>
      <StyledBreadcrumb__Item>
        <Link to="/">{categoryName}</Link>
      </StyledBreadcrumb__Item>
    </StyledBreadcrumb>
  );
}

export default Breadcrumb;
