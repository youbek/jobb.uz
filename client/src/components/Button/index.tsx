import styled, { css } from "styled-components";

type Color = "primary" | "secondary" | "outline";

interface Props {
  color?: Color;
}

const outlineStyle = css`
  color: #5b5e64;
  border: 1px solid #cbd1d4;
  background-color: #fff;

  &:hover {
    background-color: #fff;
  }
`;

const primaryStyle = css`
  background-color: #f64f64;

  &:hover {
    background-color: #db495b;
  }
`;

const secondaryStyle = css`
  background-color: #4687ff;

  &:hover {
    background-color: #3669c6;
  }
`;

const Button = styled.button<Props>`
  display: inline-block;
  font-weight: 600;
  border: 0;
  text-align: center;
  vertical-align: middle;
  user-select: none;
  padding: 10px 16px;
  font-size: 14px;
  color: #fff;
  line-height: 1.5;
  text-decoration: none;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;

  ${primaryStyle};

  ${({ color }) => color === "outline" && outlineStyle}
  ${({ color }) => color === "secondary" && secondaryStyle}
`;

export default Button;
