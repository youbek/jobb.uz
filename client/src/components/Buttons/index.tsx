import React from "react";
import styled from "styled-components";

function Button({ outline, link }) {
  return <button></button>;
}

export default Button;
