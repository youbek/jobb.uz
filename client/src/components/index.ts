// LAYOUT
export { default as Col } from "./Layout/Col";
export { default as Container } from "./Layout/Container";
export { default as Row } from "./Layout/Row";
export { default as OverlayContainer } from "./Layout/OverlayContainer/OverlayContainer";
export { default as OverlayContainerFooter } from "./Layout/OverlayContainer/OverlayContainerFooter";
export { default as Breadcrumb } from "./Breadcrumb";

// BUTTONS
export { default as Button } from "./Buttons/Button";
export { default as ButtonLink } from "./Buttons/ButtonLink";

// FORMS
export { default as CustomCheckbox } from "./Form/CustomCheckbox";
export { default as Input } from "./Form/Input";
export { default as InputGroup } from "./Form/InputGroup";
export { default as FormInline } from "./Form/FormInline";
export { default as Label } from "./Form/Label";
export { default as Select } from "./Form/Select";

// JOB RELATED
export { default as VacancyCard } from "./VacancyCard";
export { default as JobCardBadges } from "./JobsFeed/JobCard/JobCardBadges";
export { default as JobDescription } from "./JobsFeed/JobCard/JobDescription";
export { default as JobsFilterMobile } from "./JobsFeed/JobsFilter/JobsFilterMobile";

// HEADER
export { default as NavbarNav } from "./Navbar/NavbarNav";
export { default as NavContainer } from "./Navbar/NavContainer";
export { default as NavBrand } from "./Navbar/NavBrand";
export { default as NavItem } from "./Navbar/NavItem";
export { default as NavLink } from "./Navbar/NavLink";
export { default as Navbar } from "./Navbar/Navbar";

export { default as ErrorBoundary } from "./ErrorBoundary";

// Job Page
