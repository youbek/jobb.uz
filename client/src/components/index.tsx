export { default as Header } from "./Header";
export { default as Col } from "./Layout/Col";
export { default as Container } from "./Layout/Container";
export { default as Row } from "./Layout/Row";
export { default as Overlay } from "./Overlay";
export { default as Breadcrumb } from "./Breadcrumb";
export { default as VacancyFeed } from "./VacancyFeed";
export { default as VacancyFilters } from "./VacancyFilters";
export { default as Badge } from "./Badge";
export { default as Spinner } from "./Spinner";
export { default as Alert } from "./Alert";
export { default as Helmet } from "./Helmet";
export { default as Button } from "./Button";
export { default as Checkbox } from "./Form/Checkbox";
export { default as Input } from "./Form/Input";
export { default as Label } from "./Form/Label";
export { default as VacancyCard } from "./VacancyCard";
export { default as ErrorBoundary } from "./ErrorBoundary";
