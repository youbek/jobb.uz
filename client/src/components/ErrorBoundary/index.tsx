import React, { Component, ErrorInfo } from "react";
import { Container } from "components";

interface Props extends React.ComponentClass {}

interface State {
  hasError: boolean;
}

class ErrorBoundry extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error: Error) {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.log(error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      return (
        <Container className="text-center mt-4">
          <h3>Something Went Wrong</h3>
          <a href="/">Return Home</a>
        </Container>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundry;
