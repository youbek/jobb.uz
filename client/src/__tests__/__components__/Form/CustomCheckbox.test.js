import React from "react";
import { render } from "@testing-library/react";

import { CustomCheckbox } from "components";

test("CustomCheckbox rendering", () => {
  const { container: checkBox } = render(
    <CustomCheckbox defaultChecked={true} />,
  );

  expect(checkBox.firstChild.getAttribute("type")).toBe("checkbox");

  expect(checkBox.firstChild).toMatchSnapshot();
});
