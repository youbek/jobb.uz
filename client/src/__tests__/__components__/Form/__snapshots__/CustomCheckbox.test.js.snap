// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCheckbox rendering 1`] = `
<input
  checked=""
  class="CustomCheckbox-sc-19rzvd-0 hrgxbd"
  type="checkbox"
/>
`;
